runtime/java_test.go

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright (C) 2017 XLAB, Ltd.
+ *
+ * This work is open source software, licensed under the terms of the
+ * BSD license as described in the LICENSE file in the top-level directory.
+ */
+
+package runtime
+
+import (
+	"testing"
+
+	. "github.com/mikelangelo-project/capstan/testing"
+	. "gopkg.in/check.v1"
+	"gopkg.in/yaml.v2"
+)
+
+// Hook up gocheck into the "go test" runner.
+func Test(t *testing.T) { TestingT(t) }
+
+type javaSuite struct {
+}
+
+var _ = Suite(&javaSuite{})
+
+func (*javaSuite) TestGetBootCmd(c *C) {
+	// Simulate openjdk8-zulu-compact1's meta/run.yaml being parsed.
+	cmdConfs := map[string]*CmdConfig{
+		"openjdk8-zulu-compact1": &CmdConfig{
+			RuntimeType:      Native,
+			ConfigSetDefault: "java",
+			ConfigSets: map[string]Runtime{
+				"java": nativeRuntime{
+					BootCmd: "/java.so",
+					CommonRuntime: CommonRuntime{
+						Env: map[string]string{
+							"XMS":       "512m",
+							"XMX":       "512m",
+							"CLASSPATH": "/",
+							"JVM_ARGS":  "-Duser.dir=/",
+							"MAIN":      "main.Hello",
+							"ARGS":      "",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	m := []struct {
+		comment      string
+		runYamlText  string
+		expectedBoot string
+		expectedEnv  []string
+	}{
+		{
+			"simple",
+			`
+			runtime: java
+			config_set:
+			  default:
+			    main: demo.Main
+			    classpath:
+			      - /src
+			`,
+			"/java.so", []string{
+				"--env=XMS?=512m",
+				"--env=XMX?=512m",
+				"--env=CLASSPATH?=/src",
+				"--env=JVM_ARGS?=-Dx=y",
+				"--env=MAIN?=demo.Main",
+				"--env=ARGS?=",
+			},
+		},
+		{
+			"multiple classpath",
+			`
+			runtime: java
+			config_set:
+			  default:
+			    main: demo.Main
+			    classpath:
+			      - /src
+			      - /src/package1
+			`,
+			"/java.so", []string{
+				"--env=XMS?=512m",
+				"--env=XMX?=512m",
+				"--env=CLASSPATH?=/src:/src/package1",
+				"--env=JVM_ARGS?=-Dx=y",
+				"--env=MAIN?=demo.Main",
+				"--env=ARGS?=",
+			},
+		},
+		{
+			"xms and xmx",
+			`
+			runtime: java
+			config_set:
+			  default:
+			    main: demo.Main
+			    classpath:
+			      - /src
+			    xms: 128m
+			    xmx: 1024m
+			`,
+			"/java.so", []string{
+				"--env=XMS?=128m",
+				"--env=XMX?=1024m",
+				"--env=CLASSPATH?=/src",
+				"--env=JVM_ARGS?=-Dx=y",
+				"--env=MAIN?=demo.Main",
+				"--env=ARGS?=",
+			},
+		},
+		{
+			"jvm args",
+			`
+			runtime: java
+			config_set:
+			  default:
+			    main: demo.Main
+			    classpath:
+			      - /src
+			    jvm_args:
+			      - -Darg1=val1
+			      - -Darg2=val2
+			`,
+			"/java.so", []string{
+				"--env=XMS?=512m",
+				"--env=XMX?=512m",
+				"--env=CLASSPATH?=/src",
+				"--env=JVM_ARGS?=-Darg1=val1 -Darg2=val2",
```

Comment:
Double checking: [this](https://github.com/cloudius-systems/osv/issues/892) is still an issue for this case?